sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -53,12 +54,54 @@ class GpuGenerateExecSparkPlanMeta(
   }
 
   override def convertToGpu(): GpuExec = {
-    GpuGenerateExec(
-      childExprs.head.convertToGpu().asInstanceOf[GpuGenerator],
-      gen.requiredChildOutput,
-      gen.outer,
-      gen.generatorOutput,
-      childPlans.head.convertIfNeeded())
+    // if child expression contains Stack, use GpuExpandExec instead
+    if (childExprs.exists(_.isInstanceOf[GpuStackMeta])) {
+      val stackMeta = childExprs.filter(_.isInstanceOf[GpuStackMeta]).head
+      val numRows = stackMeta.childExprs.head.convertToGpu()
+          .asInstanceOf[GpuLiteral].value.asInstanceOf[Int]
+      val numFields = Math.ceil((stackMeta.childExprs.length - 1.0) / numRows).toInt
+      val projections: Seq[Seq[Expression]] = for (row <- 0 until numRows) yield {
+        val childExprsGpu = childExprs.filterNot(_.isInstanceOf[GpuStackMeta])
+            .map(_.convertToGpu())
+        val a:Seq[Expression] = for (req <- childExprsGpu) yield {
+          req
+        }
+        val b:Seq[Expression] = for (col <- 0 until numFields) yield {
+          val index = row * numFields + col
+          if (index >= stackMeta.childExprs.tail.length) {
+            GpuLiteral(null, NullType)
```

Comment:
This cannot be a NullType. It needs to be the same type as the data that will ultimately end up in the same column as it. Otherwise we are going to get some errors.